sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientException.java

```diff
@@ -248,6 +250,15 @@ CosmosClientException setCosmosResponseDiagnostics(CosmosResponseDiagnostics cos
         return this;
     }
 
+    public RequestTimeline getRequestTimeline() {
```

Comment:
This is public surface area change, so we need to make sure if we know this is applicable to all connection modes or RNTBD only.

a few questions.
1. is this related to only direct stack? what about GW mode?
2. is there any reason this is not added to request diagnostics `CosmosResponseDiagnostics` which already exists? `CosmosClientException` has a getter for it`CosmosClientException::getCosmosResponseDiagnostics()`

Reply:
This should be usable in all connection modes. It is our choice. I expected it would also be used by the HttpTransportClient and wrote a little about that here: #6580.

Reply:
Logged issue #7342, [FEATURE REQ] Add request timeline to HttpClient and HttpTransportClient

Reply:
David, what about the send portion of my comment?

> is there any reason this is not added to request diagnostics CosmosResponseDiagnostics which already exists? CosmosClientException has a getter for itCosmosClientException::getCosmosResponseDiagnostics()

Reply:
That could easily be done and I'll plan on making the change in the next few minutes. Let me touch base with Naveen before completing the work.

Reply:
Hi Mo , the best possible solution will only come once i start integrating it in my next PR , so at this point it will be hard to find the best place for RequestTimeline. For now lets keep it as David did, and I will change in my PR which I already start working on

Reply:
sure. makes sense.